src/Microsoft.CodeQuality.Analyzers/UnitTests/Maintainability/RemoveUnusedLocalsTests.Fixer.cs

```diff
@@ -30,5 +32,484 @@ protected override CodeFixProvider GetCSharpCodeFixProvider()
+        {
+            var code = @"
+class C
+{
+    int M()
+    {
+        int a = 0, b = 0;
+        return a;
+    }
+}
+";
+            var fix = @"
+class C
+{
+    int M()
+    {
+        int a = 0;
+        return a;
+    }
+}
+";
+            VerifyCSharpFix(code, fix, allowNewCompilerDiagnostics: true);
+            VerifyCSharpFixAll(code, fix, allowNewCompilerDiagnostics: true);
+        }
+
+        [Fact]
+        public void UnusedLocal_JointDeclaration_RemoveBoth_CSharp()
+        {
+            var code = @"
+class C
+{
+    int M()
+    {
+        int a = 0, b = 0;
+        return 0;
+    }
+}
+";
+            var fix = @"
+class C
+{
+    int M()
+    {
+        return 0;
+    }
+}
+";
+            VerifyCSharpFix(code, fix, allowNewCompilerDiagnostics: true);
+            VerifyCSharpFixAll(code, fix, allowNewCompilerDiagnostics: true);
+        }
+
+        [Fact(Skip ="https://github.com/dotnet/roslyn/issues/23322")]
+        public void UnusedLocal_JointAssignment_RemoveFirst_CSharp()
+        {
+            var code = @"
+class C
+{
+    int M()
+    {
+        int a = 0;
+        int b = 0;
+        a = b = 0;
+        return b;
+    }
+}
+";
+            var fix = @"
+class C
+{
+    int M()
+    {
+        int b = 0;
+        b = 0;
+        return b;
+    }
+}
+";
+            VerifyCSharpFix(code, fix, allowNewCompilerDiagnostics: true);
+            VerifyCSharpFixAll(code, fix, allowNewCompilerDiagnostics: true);
+        }
+
+        [Fact]
+        public void UnusedLocal_JointAssignment_RemoveSecond_CSharp()
+        {
+            var code = @"
+class C
+{
+    int M()
+    {
+        int a = 0;
+        int b = 0;
+        a = b = 0;
+        return a;
+    }
+}
+";
+            var fix = @"
+class C
+{
+    int M()
+    {
+        int a = 0;
+        a = 0;
+        return a;
+    }
+}
+";
+            VerifyCSharpFix(code, fix, allowNewCompilerDiagnostics: true);
+            VerifyCSharpFixAll(code, fix, allowNewCompilerDiagnostics: true);
+        }
+
+        [Fact]
+        public void UnusedLocal_JointAssignment_RemoveBoth_CSharp()
+        {
+            var code = @"
+class C
+{
+    int M()
+    {
+        int a = 0;
+        int b = 0;
+        a = b = 0;
+        return 0;
+    }
+}
+";
+            var fix = @"
+class C
+{
+    int M()
+    {
+        return 0;
+    }
+}
+";
+            // fixAll removes unused locals found in a single iteration.
+            // a is used in the first iteration.
+            var fixAll = @"
+class C
+{
+    int M()
+    {
+        int a = 0;
+        a = 0;
+        return 0;
+    }
+}
+";
+            VerifyCSharpFix(code, fix, allowNewCompilerDiagnostics: true);
+            VerifyCSharpFixAll(code, fixAll, allowNewCompilerDiagnostics: true);
+        }
+
+        [Fact]
+        public void UnusedLocal_LocalFunction_CSharp()
+        {
+            var code = @"
+class C
+{
+    int M()
+    {
+        int L() { int a = 1; return 1; }
```

Comment:
Consider adding another unit test where the body of the local function uses a local declared/assigned earlier in the method, and removal of local function makes that local unused.